File: src/com/aslan/module/cipher/algorithms/AbstractAlgorithm.js
import Algorithm from "../Algorithm.java";

class AbstractAlgorithm extends Algorithm {

	N = null;
	R = null;
	H = null;
	BOX = null;

	init(option) {
		this.BOX = option.BOX;
		this.N = option.N;
		this.R = option.R;
		this.H = this.N >> 1;
	}
}

export default AbstractAlgorithm;

File: src/com/aslan/module/cipher/algorithms/AbstractAlgorithm.java
package com.aslan.module.cipher.algorithms;

import com.aslan.module.cipher.Algorithm;
import com.aslan.module.cipher.AlgorithmInfo;
import com.aslan.module.core.Locker;

public abstract class AbstractAlgorithm implements Algorithm {

    protected int N;
    protected int R;
    protected byte[] S;
    protected byte[] $;
    private Locker locker = new Locker();

    public void init(AlgorithmInfo option) {
        N = option.N;
        R = option.R;
    }

    @Override
    public boolean lock() {
        return locker.lock();
    }

    @Override
    public void unlock() {
        locker.unlock();
    }
}
